proposals/0000-std-experimental.rst

```diff
@@ -0,0 +1,177 @@
+Add -experimental flag
+======================
+
+.. author:: Moritz Angermann, Julian Ospald
+.. date-accepted::
+.. ticket-url::
+.. implemented::
+.. highlight:: haskell
+.. header:: This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/617>`_.
+.. sectnum::
+.. contents::
+
+Introduce a new compiler flag ``-experimental`` which guards experimental
+features. We distinguish between experimental and non-experimental features by
+their respective breaking change processes. Experimental features may introduce
+breaking changes without prior warning at any time. Non-experimental (stable)
+features on the other hand would need to follow a change evolution/deprecation
+process and could only under exceptional circumstances introduce breaking
+changes.
```

Comment:
This seems to bury the lede! The proposal says it is about adding a new flag, but in fact it is about instituting a much more stringent change process for a large fraction of GHC development work.

Reply:
This flag gives us a binary distinction between what _is_ experimental and what is not. The exact process of lifecycle would be governed by e.g. #601 or similar proposals.

Reply:
After writing the other comments further down. Let me stress that this explicitly permits to have _experimental_ features in the compiler that can (and are permitted to) undergo _rapid_ evolution with breaking changes at any point.